src/index.js

```diff
@@ -86,7 +86,14 @@ export async function run() {
 
     // Work around potential `set -e` commands in `~/.profile` (looking at you, `setup-miniconda`!)
     await execShellCommand(`echo 'set +e' >/tmp/tmate.bashrc`);
-    const setDefaultCommand = `set-option -g default-command "bash --rcfile /tmp/tmate.bashrc" \\;`;
+    let setDefaultCommand = `set-option -g default-command "bash --rcfile /tmp/tmate.bashrc" \\;`;
+
+    if (core.getInput("tmate-server-host") !== "") {
+      setDefaultCommand = `${setDefaultCommand} set-option -g tmate-server-host "${core.getInput("tmate-server-host")}\;"`;
```

Comment:
Sorry for not spotting this earlier: I am fairly certain that the backslash needs to be doubled (see the initial definition of `setDefaultCommand`).

Also, this patch introduces the first time we use input values verbatim, so I think they need to be validated. For example, the host name should only contain alphanumerical characters plus `-` plus `.`. The need for validation arises from the fact that a value could contain a double quote and a semicolon and then potentially insert code that is unchecked. Probably not a _huge_ risk, but I'd rather err on the safe side.

@mxschmitt what's your take?

Reply:
okay, I will add some validation.

My plan is...

- `tmate-server-port`: Input value can `parseInt()` and it in valid port range.
- `tmate-server-*-fingerprint`: Input value has prefix that `SHA256:`

What do you think about validation for `host`?
`host` has many patterns. e.g. Global domain (`example.com`), without TLD (`ssh-server` can resolve with internal name server), IP Address, etc...
If you want to be very strong validation, actions will send a packet to tmate-server. tmate-server response string `SSH-2.0-tmate`. 

But I have doubts that actions need strong validation.
Please tell me do you think about it.

Reply:
> Sorry for not spotting this earlier: I am fairly certain that the backslash needs to be doubled (see the initial definition of setDefaultCommand).

This patch (with a single backslash) is fine work in my environments, I also check this.

Reply:
I implemented a validator for `port` and `fingerprint` https://github.com/mxschmitt/action-tmate/pull/93/commits/bde7943bfacd9ef30380090ba7ad070ebf3f3c5e

Reply:
Hmm. I would prefer something like this:

```js
const getValidatedInput => async (key) {
  const value = core.getInput(key);
  if (!value.matches(/^[-.+A-Za-z0-9]*$/) {
    throw new Error(`Invalid value for '${key}': '${value}'`);
  }
  return value;
}
```

And then

```js
setDefaultCommand = `${setDefaultCommand} set-option -g tmate-server-host "${await getValidatedInput("tmate-server-host")}\\;"`;
```

(Note also the double backslash before the semicolon.)

This should catch invalid values, even if it does not _completely_ validate them _for use with `tmate`_.

Reply:
Oh, I just realized that we do not even need to use `async`/`await` here...

Reply:
> 
> 
> > Sorry for not spotting this earlier: I am fairly certain that the backslash needs to be doubled (see the initial definition of setDefaultCommand).
> 
> This patch (with a single backslash) is fine work in my environments, I also check this.

Either this is wrong in the patch, or in the existing line above the added code block. I do not want to have mixed `\\;` and `\;`.

Reply:
@dscho Thanks for your comments and sample code!
I fixed validate function and used only `\\;` in https://github.com/whywaita/action-tmate/commit/bae5c138f677c72ea0795832f3c41aca114f2a5e.